packages/design-system/src/components/ButtonIcon/variations/ButtonIcon.tsx

```diff
@@ -0,0 +1,15 @@
+import React, { forwardRef, Ref } from 'react';
+import ButtonIconPrimitive, { DefaultTypes } from '../Primitive/ButtonIconPrimitive';
+
+export type ButtonIconType = Omit<DefaultTypes, 'variant'>;
+
+const ButtonIcon = forwardRef((props: ButtonIconType, ref: Ref<HTMLButtonElement>) => {
+	const { children, ...rest } = props;
+	return (
+		<ButtonIconPrimitive {...rest} variant="default" ref={ref}>
+			{children}
+		</ButtonIconPrimitive>
+	);
+});
```

Comment:
spreading props seems useless

```suggestion
const ButtonIcon = forwardRef((props: ButtonIconType, ref: Ref<HTMLButtonElement>) => {
	return (
		<ButtonIconPrimitive {...props} variant="default" ref={ref} />
	);
});
```

Reply:
You'll get yelled at by TS for not using `children` "the right way" :D